linux/gnome/src/main/java/com/github/weisj/darkmode/platform/linux/gnome/GnomeSettings.kt

```diff
@@ -0,0 +1,98 @@
+package com.github.weisj.darkmode.platform.linux.gnome
+
+import com.github.weisj.darkmode.platform.LibraryUtil
+import com.github.weisj.darkmode.platform.settings.*
+import com.google.auto.service.AutoService
+import kotlin.streams.toList
+
+/**
+ * Workaround for the fact that auto service currently doesn't work with singleton objects.
+ * https://github.com/google/auto/issues/785
+ */
+@AutoService(SettingsContainer::class)
+class GnomeSettingsProxy : SettingsContainer by GnomeSettings
+
+fun <T> concatenate(vararg lists: List<T>): List<T> {
+    val result: MutableList<T> = ArrayList()
+    lists.forEach { list: List<T> -> result.addAll(list) }
+    return result
+}
+
+object GnomeSettings : DefaultSettingsContainer() {
+    override val enabled: Boolean
+        get() = super.enabled && LibraryUtil.isGnome
+
+    /**
+     * This enum holds default values for the light, dark, and high contrast GTK themes.
+     * The defaults are a safe bet as they are included with almost any install of GTK.
+     * Even if they're not present, the way the logic works in GnomeThemeMonitorService,
+     * there won't be an issue and the plugin will fall back on the light theme.
+     *
+     * **See:** [Arch Wiki on GTK](https://wiki.archlinux.org/index.php/GTK#Themes)
+     */
+    private enum class DefaultGtkTheme(val info: GtkTheme) {
+        DARK(GtkTheme("Adwaita-dark")),
+        LIGHT(GtkTheme("Adwaita")),
+        HIGH_CONTRAST(GtkTheme("HighContrast")),
+    }
+
+    private const val DEFAULT_GUESS_LIGHT_AND_DARK_THEMES = true
+
+    @JvmField
+    var darkGtkTheme = DefaultGtkTheme.DARK.info
+
+    @JvmField
+    var lightGtkTheme = DefaultGtkTheme.LIGHT.info
+
+    @JvmField
+    var highContrastGtkTheme = DefaultGtkTheme.HIGH_CONTRAST.info
+
+    @JvmField
+    var guessLightAndDarkThemes = DEFAULT_GUESS_LIGHT_AND_DARK_THEMES
+
+    init {
+        if (!GnomeLibrary.get().isLoaded) {
```

Comment:
Currently setting containers are always created. This will throw the exception on all machines that don't use Gnome. I'll move container provisioning to make use of a `SettingsContainerProvider` service. This way only settings that are actually suitable for a machine are created.

Something to consider: Only settings of containers that are actually created and enabled are stored. Because settings are stored on an OS basis this isn't an issue currently but will become one once there are more than one implementations for linux.

Reply:
Should I then change the `GnomeSettingsProvider` declaration to be `class GnomeSettingsProvider : SingletonSettingsContainerProvider({ GnomeSettings }, LibraryUtil.isGnome)`?

Reply:
Yes exactly. For the purpose of readability though I would go for `enabled = LibraryUtil.isGnome`.

Reply:
c44ab2f7ab7970cf5cc10a84622ace3868e3f2ba